gammapy/spectrum/spectrum_grouping.py

```diff
@@ -48,17 +47,21 @@ def group_obs_table(obs_table, offset_range=[0, 2.5], n_off_bin=5,
     offmin, offmax = offset_range
     effmin, effmax = eff_range
     zenmin, zenmax = zen_range
-
+    coszenmin = np.cos(zenmax * np.pi / 180.)
+    coszenmax = np.cos(zenmin * np.pi / 180.)
     offtab = Angle(np.linspace(offmin, offmax, n_off_bin + 1), 'deg')
-    efftab = Quantity(np.linspace(effmin, effmax, n_eff_bin + 1)/100., '')
-    coszentab = Quantity(np.linspace(zenmin, zenmax, n_zen_bin + 1), '')
+    efftab = Quantity(np.linspace(effmin, effmax, n_eff_bin + 1) / 100., '')
+    coszentab = Quantity(np.linspace(coszenmin, coszenmax, n_zen_bin + 1), '')
 
     val = list()
     val.append(ObservationGroupAxis('MUONEFF', efftab, 'edges'))
-    val.append(ObservationGroupAxis('ZEN_PNT', coszentab, 'edges'))
+    val.append(ObservationGroupAxis('COSZEN', coszentab, 'edges'))
     val.append(ObservationGroupAxis('OFFSET', offtab, 'edges'))
 
     obs_groups = ObservationGroups(val)
+    import IPython; IPython.embed()
+    cos_zen = np.cos(obs_table['ZEN_PNT'].quantity.value * np.pi / 180.)
```

Comment:
Hi,
For this remark the problem is that in the obs-index table of hap-FR or in yours for example in gammapy-extra/datasets/hess-crab4-hd-hap-prod2, the ZEN_PNT column is not in degrees.
I wanted to use np.cos(obs_table['ZEN_PNT'].quantity) but I can't since there is no unit... I tested with data from PA where the ZEN_PNT column is an angle and the np.cos(obs_table['ZEN_PNT']) without .quantity gives creasy result.

The problem comes from the fact that in HAP-FR or HAP-HD to create the obs-index table and the column ZEN_PNT we use the column ALT_PNT that is not an angle in the events fits file. Maybe when we create the obs-index table we have to force the ZEN_PNT column to be an angle and then use np.cos(obs_table['ZEN_PNT'].quantity)? What do you think? It's just in your exporter that you have to changed that.